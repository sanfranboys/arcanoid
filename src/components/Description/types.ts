export type DescriptionProps = {
  title: string,
  noBorder?: boolean
}
