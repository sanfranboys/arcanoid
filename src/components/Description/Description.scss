.description {
  border-bottom: 1px solid var(--color-border-dark);
  padding: 5px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  color: var(--lightBg);

  &_border_no {
    border-bottom: none;
  }

  &__key {
    font-size: var(--size-text-s);

  }

  &__value {
    font-size: var(--size-text-l);
  }
}
