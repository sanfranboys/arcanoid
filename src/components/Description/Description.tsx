import React, { FC } from 'react';
import { DescriptionProps } from './types';
import classNames from '../../utils';

import './Description.scss';

const Description: FC<DescriptionProps> = (props) => {
  const { title, children, noBorder } = props;
  const className = classNames('description', { 'description_border_no': noBorder })

  return (
    <div className={className}>
      <div className='description__key'>{ title }</div>

      <div className='description__value'>
        { children }
      </div>
    </div>
  )
}

export default Description;
