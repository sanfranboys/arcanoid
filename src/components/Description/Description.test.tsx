import React from 'react'
import { create } from 'react-test-renderer'
import Description from './Description'

describe('Description testing', () => {
  const props = {
    title: 'test-title',
    noBorder: true,
    children: 'Content here',
  }
  describe('Description snapshot testing', () => {
    it('snapshot matches', async () => {
      const tree = create(<Description {...props} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
