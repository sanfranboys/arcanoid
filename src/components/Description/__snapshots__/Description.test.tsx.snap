// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description testing Description snapshot testing snapshot matches 1`] = `
<div
  className="description description_border_no"
>
  <div
    className="description__key"
  >
    test-title
  </div>
  <div
    className="description__value"
  >
    Content here
  </div>
</div>
`;
